experimental/coref/scripts/prep_span_data.py

```diff
@@ -0,0 +1,170 @@
+)
+parser.add_argument(
+    "--span-prefix", help="Prefix in the doc.spans used to store span-clusters."
+)
+parser.add_argument("--limit", type=int, help="Number of documents to process.")
+parser.add_argument(
+    "--heads",
+    type=str,
+    default="silver",
+    help="Whether to use gold heads or silver heads predicted by the clustering component",
+)
+parser.add_argument(
+    "--gpu", type=int, default=-1, help="ID of GPU to run coreference pipeline on."
+)
+
+args = parser.parse_args()
+
+
+def find_target_span(head, ex):
+    """
+    Take the smallest enclosing gold-span as
+    the target for each predicted head.
+    """
+
+    # Note: This is slow because it doesn't assume that the smallest enclosing
+    # span for a given token will necessarily be in the matching spangroup. For
+    # example, a token could be in word-level cluster 1, but the smallest
+    # enclosing span could be in cluster 2. When using word-level tokens
+    # predicted by a trained model, like in this script, you have to check
+    # because there's no guarantee tokens and spans will align.
+
+    # If you know your spangroups are aligned, you could just check the
+    # matching spangroup. However that still wouldn't guarantee you get the
+    # smallest enclosing span, since it is possible (though rare/unlikely) for
+    # the same head to have different spans associated with it in different
+    # spangroups.
+
+    smallest = float("inf")
+    target_span = None
+    for name, span_group in ex.reference.spans.items():
+        if name.startswith(args.span_prefix):
+            aligned_spans = ex.get_aligned_spans_y2x(span_group)
+            for span in aligned_spans:
+                if span.start <= head.start and head.end <= span.end:
+                    size = span.end - span.start
+                    if size == 1:
+                        # won't get smaller, short ciruit
+                        return span
+                    if size < smallest:
+                        target_span = span
+                        smallest = size
+
+    return target_span
+
+
+if args.gpu > -1:
+    spacy.require_gpu(args.gpu)
+nlp = spacy.load(args.model_path)
+docbin = DocBin().from_disk(args.input_path)
+output_docbin = DocBin()
+docs = docbin.get_docs(nlp.vocab)
+input_head_clusters = {}
+target_span_clusters = {}
+total_heads = 0
+duplicate_heads = 0
+kept_heads = 0
+empty_clusters = 0
+num_docs = 0
+skipped_docs = 0
+
+# Gold heads can be used directly from the input data, or heads predicted by
+# the clustering component can be used. In tests with OntoNotes, accuracy and
+# processing time were similar for both options. However, silver heads avoid
+# tokenization mismatches, so they are chosen as the default.
+use_gold_heads = False
+if args.heads == "silver":
+    use_gold_heads = False
+elif args.heads == "gold":
+    use_gold_heads = True
+else:
+    raise ValueError(f"Expected 'gold' or 'silver' for heads but got: {args.heads}")
```

Comment:
I'd avoid the indirect `if`-`else` logic to raise the error, and instead I'd write
```suggestion
if args.heads not in ["silver", "gold"]:
    raise ValueError(f"Expected 'gold' or 'silver' for heads but got: {args.heads}")
use_gold_heads = False
if args.heads == "gold":
    use_gold_heads = True
```